lightning/src/util/events.rs

```diff
@@ -128,6 +129,12 @@ pub enum Event {
 		/// the payment has failed, not just the route in question. If this is not set, you may
 		/// retry the payment via a different route.
 		rejected_by_dest: bool,
+		/// Any failure information conveyed via the Onion return packet by a node along the failed
+		/// payment route. Should be applied to the [`NetworkGraph`] so that routing decisions can
+		/// take into account the update.
+		///
+		/// [`NetworkGraph`]: crate::routing::network_graph::NetworkGraph
+		network_update: Option<NetworkUpdate>,
```

Comment:
Would be good to describe how to apply the event to the NetworkGraph and that BackgroundProcessor will do this for you